File: GC.Domain/Contacts/ForeignContacts/ForeignContactType.cs
using System;

namespace GC.Domain.Contacts.ForeignContacts
{
    public enum ForeignContactType
    {
        /// <summary>
        /// Элекрик
        /// </summary>
        Electric = 1,

        /// <summary>
        /// Участковый
        /// </summary>
        LocalPoliceman = 2,

        /// <summary>
        /// Лесник
        /// </summary>
        Forester = 3
    }

    public static class ForeignContactTypeExtensions
    {
        public static String GetDisplayName(this ForeignContactType type)
        {
            switch(type)
            {
                case ForeignContactType.Electric: return "Старший участковый лесничий";
                case ForeignContactType.LocalPoliceman: return "Участковый";
                case ForeignContactType.Forester: return "Электрик";

                default: throw new Exception("Точка недостижимости");
            }
        }
    }
}

File: GC.WebSpace/ClientApp/domain/contacts/foreignContacts/foreignContactType.ts
import { Never } from "../../../tools/never";

export enum ForeignContactType {
    Electric = 1,
    LocalPoliceman = 2,
    Forester = 3
}

export namespace ForeignContactType {
    export function getDisplayName(type: ForeignContactType): string {
        switch (type) {
            case ForeignContactType.Electric: return "Электрик";
            case ForeignContactType.LocalPoliceman: return "Участковый";
            case ForeignContactType.Forester: return "Старший лесничий";

            default: throw new Never(type);
        }
    }
}